Show "Standard input code" instead of "-" in error messages

TazeTSchnitzel opened the pull request:
Say you have broken code, like this:

``` php
<?php
declare(strict_types=1);
function add(int $a, int $b): int {
    return $a + $b;
}
add(1.0, 2.0);
```

Currently, the error message looks like this:

```
Fatal error: Uncaught TypeException: Argument 1 passed to add() must be of the type integer, float given, called in - on line 6 and defined in -:3
Stack trace:
#0 -(6): add(1, 2)
#1 {main}
  thrown in - on line 3
```

What on earth is `-`? It may not be immediately obvious to the user.

So, this makes them look like this:

```
Fatal error: Uncaught TypeException: Argument 1 passed to add() must be of the type integer, float given, called in php://stdin on line 6 and defined in php://stdin:3
Stack trace:
#0 php://stdin(6): add(1, 2)
#1 {main}
  thrown in php://stdin on line 3
```

I think `php://stdin` is a lot more obvious. It's a real URL too, [at least in PHP](http://php.net/manual/en/wrappers.php.php).

TazeTSchnitzel commented:
Any chance of this being merged?

TazeTSchnitzel commented:
Name might be misleading given you can't actually get that input from `php://stdin`.

Maybe it should be called “Standard input code” instead? That would match “Command line code”.

Tyrael commented:
I like the second name better, but this wasn't the reason for not being merged, I've just missed it tbh.

TazeTSchnitzel commented:
Rebased against current master (to resolve merge conflict), and added a second patch to change to the friendlier "Standard input code" wording. If this is merged, it should be squashed together.

I could merge this myself, of course, but I'd rather not do that if it would be controversial, hence the pull request.

TazeTSchnitzel commented:
This needs updating for phpdbg.

KalleZ commented:
@TazeTSchnitzel just commit it to 7.2, afterwards 7.1 is out both @dshafik and @krakjoe can consider down merging it, so we do not push too many things on to 7.1 :)

krakjoe commented:
Merged into 7.2 for now, updated phpdbg ...

@TazeTSchnitzel remind us to look at this once 7.1 is GA as suggested by Kalle.

TazeTSchnitzel commented:
Wait, are you suggesting merging it into 7.1 once it's released?!

KalleZ commented:
@TazeTSchnitzel thats the plan yeah, or do you have something else on your mind?

TazeTSchnitzel commented:
I mean, you could do that. It feels a little late to me to do that, but I suppose this change probably can't break anything.

bwoebi commented:
Not sure whether it should be merged into 7.1 after GA. It led to subtle breaks (e.g. phpdbg assuming it's "-"), other extensions and even php scripts may be doing such assumptions too.

Either we merge it in _before_ or into 7.2 only.

krakjoe commented:
Many projects are including master in their CI, so my thinking was that we could wait to see what kind of breakages actually do appear in the wild, before making a decision either way.

It's not a big deal either way ...

TazeTSchnitzel commented:
The sort of thing @bwoebi's worried is what was on my mind.

krakjoe commented:
Had crossed mine while I was changing phpdbg, but I'm just not sure what kind of effect it really will have, outside of the obvious debuggers.

It's difficult to imagine why a user script would be relying on the old behaviour, other than by proxy, relying on xdebug/phpdbg, isn't it ?

TazeTSchnitzel commented:
Aside from people like me who are accessing opcode arrays from userland with experimental extensions, yeah, doubt it'd hit anyone.
